cpp/src/rolling/jit/kernel.cu

```diff
@@ -51,7 +51,7 @@ __global__ void gpu_rolling_new(cudf::size_type nrows,
                                 FollowingWindowType following_window_begin,
                                 cudf::size_type min_periods)
 {
-  cudf::size_type i      = blockIdx.x * blockDim.x + threadIdx.x;
+  std::size_t i          = blockIdx.x * blockDim.x + threadIdx.x;
   cudf::size_type stride = blockDim.x * gridDim.x;
```

Comment:
We should be consistent about the types of `i` and `stride`, regardless of what that type is.

Reply:
>I'd also propose introducing a type declaration here, like what we do for size_type: using thread_index_type = int64_t;. That might reduce the risk of developer mistakes and would enforce consistency for the thread index, the stride, etc.

This is an excellent idea.